code/controllers/subsystem/verb_manager.dm

```diff
@@ -0,0 +1,86 @@
+ * plus TICK_BYOND_RESERVE from the tick and uses up to that amount of time (minus the percentage of the tick used by the time it executes subsystems)
+ * on subsystems running cool things like atmospherics or Life or SSInput or whatever.
+ *
+ * Without this subsystem, verbs are likely to cause overtime if the MC uses all of the time it has alloted for itself in the tick, and SendMaps
+ * uses as much as its expected to, and an expensive verb ends up executing that tick. This is because the MC is completely blind to the cost of
+ * verbs, it can't account for it at all. The only chance for verbs to not cause overtime in a tick where the MC used as much of the tick
+ * as it alloted itself and where SendMaps costed as much as it was expected to is if the verb(s) take less than TICK_BYOND_RESERVE percent of
+ * the tick, which isnt much. Not to mention if SendMaps takes more than 30% of the tick and the MC forces itself to take at least 70% of the
+ * normal tick duration which causes ticks to naturally overrun even in the absence of verbs.
+ *
+ * With this subsystem, the MC can account for the cost of verbs and thus stop major overruns of ticks. This means that the most important subsystems
+ * like SSinput can start at the same time they were supposed to, leading to a smoother experience for the player since ticks arent riddled with
+ * minor hangs over and over again.
+ */
+SUBSYSTEM_DEF(verb_manager)
+	name = "Verb Manager"
+	wait = 1
+	flags = SS_TICKER
+	priority = FIRE_PRIORITY_DELAYED_VERBS
+	runlevels = ALL//trollface
+
+	///list of callbacks to procs called from verbs or verblike procs that were executed when the server was overloaded and had to delay to the next tick.
+	///this list is ran through every tick, and the subsystem does not yield until this queue is finished.
+	var/list/verb_queue = list()
+
+	///running average of how many verb callbacks are executed every second. used for the stat entry
+	var/verbs_executed_per_second = 0
+
+	var/FOR_ADMINS_IF_VERBS_FUCKED_immediately_execute_all_verbs = FALSE
+
+///queue a callback on a proc meant to resume processing for a verb for the next tick. returns TRUE if it was queued, FALSE otherwise.
+/datum/controller/subsystem/verb_manager/proc/queue_verb(datum/thing_to_call, proc_to_call, ...)
+	//since this queue is meant specifically to delay user input as little as possible without overrunning the tick, the callback must have usr.
+	//no, you cant queue everything up for this eat shit. if its not user input then it cant go into a queue that never yields
+	if(QDELETED(thing_to_call) \
+	|| !proc_to_call \
+	|| !ismob(usr) \
+	|| QDELING(usr) \
+	|| usr.client?.holder /*dont worry admins cant cause overtime :clueless:*/ \
+	|| FOR_ADMINS_IF_VERBS_FUCKED_immediately_execute_all_verbs \
+	|| !initialized \
+	|| !(runlevels & Master.current_runlevel))
+		return FALSE
```

Comment:
this check isn't too expensive is it?

Reply:
verbs arent expensive because thousands of them happen per tick and they add up, theyre expensive because a few verbs take on the order of milliseconds or few hundred microseconds and they happen a few times per tick (at the end). so this should never be a relevant cost